src/apps/utils/LinkedDocumentsHelper.js

```diff
@@ -1,93 +1,144 @@
+   * @returns {Boolean} `true` if removed
+   */
   async remove(uuid) {
-    const key = this._findKey(uuid);
+    const key = this.getLinkedPropertyKey(uuid);
     if (key) {
-      await this._container.update({ [`system.${this._pathKey}.-=${key}`]: null });
+      await this.#container.update({ [`system.${this.#propertyName}.-=${key}`]: null });
       return true;
     }
     return false;
   }
 
+  /**
+   * Finds the Linked Document with `uuid`
+   *
+   * @param {String} uuid
+   * @returns {Object|Null}
+   */
   find(uuid) {
-    const key = this._findKey(uuid);
-    if (key) {
-      return this.linkRefs[key];
-    }
-    return null;
+    const key = this.getLinkedPropertyKey(uuid);
+    return key ? this.linkedDocuments[key] : null;
   }
 
-  get linkRefs() {
-    return this._container.system[this._pathKey];
+  get linkedDocuments() {
+    return this.#container.system[this.#propertyName];
   }
 
-  _findKey(uuid) {
-    const entry = Object.entries(this.linkRefs).find((e) => e[1]?.uuid === uuid);
-    return entry ? entry[0] : null;
+  /**
+   * The random linkedDocuments property for element with `uuid`
+   *
+   * @param {String} uuid Foundry's item id
+   * @returns {String|Undefined}
+   */
+  getLinkedPropertyKey(uuid) {
+    const [key] = Object.entries(this.linkedDocuments).find(
+      // eslint-disable-next-line no-unused-vars
+      ([_, value]) => value.uuid === uuid
+    );
+    return key;
   }
 
-  async _add(linkRef, replaceKey) {
+  /**
+   * Tests if `linkedDocument` can added, and adds it.
+   *
+   * @param {DocumentLinkReference} linkedDocument The document reference to add
+   * @param {String=} replacementTarget The key on `this.linkedDocuments` where to insert
+   * `linkedDocument. If none is provided, it generates a new key value.
+   * @returns {Boolean} `true` if added.
+   */
+  async #add(linkedDocument, replacementTarget) {
     // Check we don't already have this feature added.
-    const duplicateIndex = this._findKey(linkRef.uuid);
-    if (duplicateIndex) {
-      ui.notifications.warn(game.i18n.localize(this._duplicateWarning));
+    const duplicate = this.getLinkedPropertyKey(linkedDocument.uuid);
+    if (duplicate) {
+      ui.notifications.warn(game.i18n.localize(this.#duplicateWarning));
       return false;
     }
 
-    if (this._validate) {
-      const obj = await fromUuid(linkRef.uuid);
-      if (!this._validate(obj)) {
-        ui.notifications.warn(game.i18n.localize(this._validateWarning));
+    if (this.#validate) {
+      const obj = await fromUuid(linkedDocument.uuid);
+      if (!this.#validate(obj)) {
+        ui.notifications.warn(game.i18n.localize(this.#validateWarning));
         return false;
       }
     }
 
-    // Generate an index if one isn't specified.
-    let key = replaceKey;
-    if (!key) {
-      key = foundry.utils.randomID();
-    }
+    // Generate an key if one isn't specified.
+    const key = replacementTarget || foundry.utils.randomID();
```

Comment:
I left a comment on this before, but your force push replaced the commit I commented on, so it seems to have been lost.

Since we're specifically providing a fallback for if `replacementTarget` is `undefined`, we should use `??` here instead of `||`. In this particular case, we're not really worried about `||` including values we don't intend, but `??` expresses our intentions a lot better.